samtranslator/model/sam_resources.py

```diff
@@ -1146,6 +1146,7 @@ class SamApi(SamResourceMacro):
         "Tags": PropertyType(False, IS_DICT),
         "DefinitionBody": PropertyType(False, IS_DICT),
         "DefinitionUri": PropertyType(False, one_of(IS_STR, IS_DICT)),
+        "MergeDefinitions": PropertyType(False, is_type(bool)),
```

Comment:
`PropertyType` is deprecated IMO. Can we use `Property` here? https://github.com/aws/serverless-application-model/blob/c5830b63857f52e540fec13b29f029458edc539a/samtranslator/model/__init__.py#L36-L45

Reply:
Yes: https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#development-guidelines

In fact this PR in its current state has a bug: if you pass an intrinsic to `MergeDefinitions` it'll happily accept it since it's using `PropertyType`, and the code checks for truthyness, so any intrinsic function that should resolve to `False` now is treated as `true`.